### ISSUE TITLE
gcc-13 error conflicting types for uct_md_mem_advice due to enum/integer mismatch

### ISSUE THREAD

**Issue by jedbrown** [pmodels/mpich#6511]
This is a `-Werror` issue that appears with 4.1.1. I mention it because it seems configure is adding the `-Werror` here, and thus has taken on this burden. 

```
../../../../../modules/ucx/src/uct/base/uct_md.c:455:1: error: conflicting types for ‘uct_md_mem_advise’ due to enum/integer mismatch; have ‘ucs_status_t(struct uct_md *, void *, void *, size_t,  unsigned int)’ {aka ‘ucs_status_t(struct uct_md *, void *, void *, long unsigned int,  unsigned int)’} [-Werror=enum-int-mismatch]
  455 | uct_md_mem_advise(uct_md_h md, uct_mem_h memh, void *addr, size_t length,
      | ^~~~~~~~~~~~~~~~~
In file included from ../../../../../modules/ucx/src/uct/base/uct_component.h:11,
                 from ../../../../../modules/ucx/src/uct/base/uct_md.h:14,
                 from ../../../../../modules/ucx/src/uct/base/uct_md.c:13:
/home/jed/src/aur/mpich/src/mpich-4.1.1/build/modules/ucx/../../../modules/ucx/src/uct/api/uct.h:2412:14: note: previous declaration of ‘uct_md_mem_advise’ with type ‘ucs_status_t(struct uct_md *, void *, void *, size_t,  uct_mem_advice_t)’ {aka ‘ucs_status_t(struct uct_md *, void *, void *, long unsigned int,  uct_mem_advice_t)’}
 2412 | ucs_status_t uct_md_mem_advise(uct_md_h md, uct_mem_h memh, void *addr,
      |              ^~~~~~~~~~~~~~~~~
```

Perhaps this patch can be applied.
```diff
diff --git i/src/uct/base/uct_md.c w/src/uct/base/uct_md.c
index 3eb58368c..23d6b89d2 100644
--- i/src/uct/base/uct_md.c
+++ w/src/uct/base/uct_md.c
@@ -453,7 +453,7 @@ ucs_status_t uct_md_mem_free(uct_md_h md, uct_mem_h memh)

 ucs_status_t
 uct_md_mem_advise(uct_md_h md, uct_mem_h memh, void *addr, size_t length,
-                  unsigned advice)
+                  uct_mem_advice_t advice)
 {
     if ((length == 0) || (addr == NULL)) {
         return UCS_ERR_INVALID_PARAM;
```

I also needed this one lest `uint32_t` is undefined.
```diff
diff --git i/test/apps/sockaddr/sa_util.h w/test/apps/sockaddr/sa_util.h
index 7cd46bcec..1cb9af5a2 100644
--- i/test/apps/sockaddr/sa_util.h
+++ w/test/apps/sockaddr/sa_util.h
@@ -11,6 +11,7 @@
 #include <sstream>
 #include <string>
 #include <vector>
+#include <cstdint>
 
 
 /* runtime error exception */

```

<details>

Configured as
```
  ../configure --prefix=/opt/mpich \
               --with-device=ch4:ucx \
               --with-hwloc \
               --without-java \
               --without-ze \
               --enable-error-checking=runtime \
               --enable-error-messages=all \
               CC=gcc CXX=g++ FC=gfortran
```
</details>

**Comment by raffenet** [comment]
I believe this is fixed by https://github.com/pmodels/mpich/pull/6468. We will pick back to 4.1.x and do a 4.1.2 probably by the end of the month.

**Comment by raffenet** [comment]
I'm closing this because I believe it to be fixed in 4.1.2 and main. Please reopen if I am wrong.
